docs/reference/migration/migrate_2_0.asciidoc

```diff
@@ -138,6 +138,9 @@ In addition some query builders have been removed or renamed:
 * `filtered(...)` removed. Use `filteredQuery(...)` instead.
 * `inQuery(...)` removed.
 
+The methods IndicesQueryBuilder#parseNoMatchQuery(String) and IndicesQueryBuilder#noMatchQuery(String) have been removed,
```

Comment:
the reason for adding MatchNone in the first place was to drop the String setter. We either need one (the new query) or the other (the string setter), not both at the same time. I think having all the parsing in the parser is slightly cleaner than having to resolve/parse strings to queries in data nodes. That's why we went for adding this new query. We can still have MatchNoneQueryBuilder public without having a parser for it, it doesn't sound bad to me. Maybe we can leave the `QueryBuilders#matchNoneQuery` method. @cbuescher what do you think about this?